lib/rack/request.rb

```diff
@@ -240,7 +245,9 @@ def xhr?
       end
 
       def host_with_port
-        if forwarded = get_header(HTTP_X_FORWARDED_HOST)
+        if forwarded = get_http_forwarder(:host).last
+          forwarded
+        elsif forwarded = get_header(HTTP_X_FORWARDED_HOST)
           forwarded.split(/,\s?/).last
```

Comment:
Curious that we use the _first_ forward for protocol, the _last_ forward for host, and the _first trusted_ forward for IP.